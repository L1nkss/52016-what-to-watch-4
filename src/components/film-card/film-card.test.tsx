import * as React from "react";
import * as renderer from "react-test-renderer";
import * as ShallowRenderer from 'react-test-renderer/shallow';
import FilmCard from "@components/film-card/film-card";
import {Router} from "react-router";
import {createMemoryHistory} from "history";

// Моки
const film = {
  name: `Fantastic Beasts: The Crimes of Grindelwald`,
  previewImage: `fantastic-beasts-the-crimes-of-grindelwald`,
  previewVideoLink: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
  id: 2
};

describe(`Testing FilmCard component`, () => {
  it(`Component should successfully rendered`, () => {
    const myShallowRenderer = ShallowRenderer.createRenderer();
    myShallowRenderer.render(<FilmCard
      filmInfo={film}
      renderPlayer={() => <video />}
      handleMouseEnter={() => {}}
      handleMouseLeave={() => {}}/>);
    const result = myShallowRenderer.getRenderOutput();
    expect(result).toMatchSnapshot();
    // const history = createMemoryHistory(`/sign-in`);
    // const component = renderer.create(
    //     <Router history={history}>
    //       <FilmCard
    //         filmInfo={film}
    //         renderPlayer={() => <video />}
    //         handleMouseEnter = {() => {}}
    //         handleMouseLeave = {() => {}}
    //       />
    //     </Router>
    // );
    // const tree = component.toJSON();
    // expect(tree).toMatchSnapshot();
  });
});
