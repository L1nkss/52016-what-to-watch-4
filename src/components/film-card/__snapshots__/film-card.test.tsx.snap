// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing FilmCard component Component should successfully rendered 1`] = `
<article
  className="small-movie-card catalog__movies-card"
  onMouseLeave={[Function]}
  onMouseOver={[Function]}
>
  <Link
    className="c-inherit"
    to="/film-detail/2"
  >
    <div
      className="small-movie-card__image"
    >
      <video />
    </div>
    <h3
      className="small-movie-card__title"
    >
      Fantastic Beasts: The Crimes of Grindelwald
    </h3>
  </Link>
</article>
`;
